src/infra/dex/balancer/dto.rs

```diff
@@ -1,58 +1,208 @@
 use {
     crate::{
-        domain::{auction, dex, order},
+        domain::{dex, eth, order},
+        infra::dex::balancer::Error,
         util::serialize,
     },
+    bigdecimal::BigDecimal,
     ethereum_types::{H160, H256, U256},
+    number::conversions::{big_decimal_to_u256, u256_to_big_decimal},
     serde::{Deserialize, Serialize},
     serde_with::serde_as,
 };
 
-#[derive(Debug, Serialize)]
+pub mod get_swap_paths_query {
+    /// Get swap quote from the SOR v2 for the V2 vault.
+    pub const QUERY: &str = r#"
+query sorGetSwapPaths($callDataInput: GqlSwapCallDataInput!, $chain: GqlChain!, $queryBatchSwap: Boolean!, $swapAmount: AmountHumanReadable!, $swapType: GqlSorSwapType!, $tokenIn: String!, $tokenOut: String!, $useVaultVersion: Int) {
+    sorGetSwapPaths(
+        callDataInput: $callDataInput,
+        chain: $chain,
+        queryBatchSwap: $queryBatchSwap,
+        swapAmount: $swapAmount,
+        swapType: $swapType,
+        tokenIn: $tokenIn,
+        tokenOut: $tokenOut,
+        useVaultVersion: $useVaultVersion
+    ) {
+        tokenAddresses
+        swaps {
+            poolId
+            assetInIndex
+            assetOutIndex
+            amount
+            userData
+        }
+        swapAmountRaw
+        returnAmountRaw
+        tokenIn
+        tokenOut
+    }
+}
+    "#;
+}
+
+#[derive(Serialize)]
 #[serde(rename_all = "camelCase")]
-pub enum OrderKind {
-    Sell,
-    Buy,
+pub struct Query<'a> {
+    query: &'a str,
+    variables: Variables,
+}
+
+impl Query<'_> {
+    pub fn from_domain(
+        order: &dex::Order,
+        slippage: &dex::Slippage,
+        chain_id: eth::ChainId,
+        contract_address: eth::ContractAddress,
+        query_batch_swap: bool,
+    ) -> Result<Self, Error> {
+        let variables = Variables {
+            call_data_input: CallDataInput {
+                deadline: None,
+                receiver: contract_address.0,
+                sender: contract_address.0,
+                slippage_percentage: slippage.as_factor().clone(),
+            },
+            chain: Chain::from_domain(chain_id)?,
+            query_batch_swap,
+            swap_amount: EtherAmount::from_wei(&order.amount.get()),
+            swap_type: SwapType::from_domain(order.side),
+            token_in: order.sell.0,
+            token_out: order.buy.0,
+            use_vault_version: VaultVersion::V2 as u8,
```

Comment:
Not sure casting this enum to u8 is 100% correct. This probably works for enums which only have a single variant. But I would expect as soon as we have multiple variants which in turn require a discriminant to be stored this might not be okay. 🤔

Reply:
I think this is only guaranteed to be correct if you use `#[repr(u8)]` on the enum. That way the compiler will complain if there are more variants than a `u8` could represent. Without that we could have 256 variants (which then require at least `u16` because we need 1 value for the discriminant) and cast the `256th` value `as u8` which would then result in `0`. The compiler would not complain that it the value now changed its meaning because conversions with `as` are allowed to be lossy.

Reply:
Updated the enum.